Google app engine deployment is failing while building package

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

<!-- Explain the bug, if an error is being thrown a stack trace helps -->

Google app engine deployment is failing while building package

[INFO] --- spring-boot-maven-plugin:2.5.8:repackage (default) @ xxxx-xxxxx ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  26.202 s
[INFO] Finished at: 2022-01-21T16:12:16+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.springframework.boot:spring-boot-maven-plugin:2.5.8:repackage (default) on project xxxx-xxxxx: Execution default of goal org.springframework.boot:spring-boot-maven-plugin:2.5.8:repackage failed: An API incompatibility was encountered while executing org.springframework.boot:spring-boot-maven-plugin:2.5.8:repackage: java.lang.AbstractMethodError: Receiver class org.springframework.boot.loader.thin.ThinLayout does not define or inherit an implementation of the resolved method 'abstract java.lang.String getLibraryLocation(java.lang.String, org.springframework.boot.loader.tools.LibraryScope)' of interface org.springframework.boot.loader.tools.Layout.
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.springframework.boot:spring-boot-maven-plugin:2.5.8
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/home/workplace/.m2/repository/org/springframework/boot/spring-boot-maven-plugin/2.5.8/spring-boot-maven-plugin-2.5.8.jar
[ERROR] urls[1] = file:/home/workplace/.m2/repository/org/springframework/boot/experimental/spring-boot-thin-layout/1.0.23.RELEASE/spring-boot-thin-layout-1.0.23.RELEASE.jar
[ERROR] urls[2] = file:/home/workplace/.m2/repository/org/springframework/boot/spring-boot-buildpack-platform/2.5.8/spring-boot-buildpack-platform-2.5.8.jar
[ERROR] urls[3] = file:/home/workplace/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.12.6/jackson-databind-2.12.6.jar
[ERROR] urls[4] = file:/home/workplace/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.12.6/jackson-annotations-2.12.6.jar
[ERROR] urls[5] = file:/home/workplace/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.12.6/jackson-core-2.12.6.jar
[ERROR] urls[6] = file:/home/workplace/.m2/repository/com/fasterxml/jackson/module/jackson-module-parameter-names/2.12.6/jackson-module-parameter-names-2.12.6.jar
[ERROR] urls[7] = file:/home/workplace/.m2/repository/net/java/dev/jna/jna-platform/5.7.0/jna-platform-5.7.0.jar
[ERROR] urls[8] = file:/home/workplace/.m2/repository/net/java/dev/jna/jna/5.7.0/jna-5.7.0.jar
[ERROR] urls[9] = file:/home/workplace/.m2/repository/org/apache/commons/commons-compress/1.21/commons-compress-1.21.jar
[ERROR] urls[10] = file:/home/workplace/.m2/repository/org/apache/httpcomponents/httpclient/4.5.13/httpclient-4.5.13.jar
[ERROR] urls[11] = file:/home/workplace/.m2/repository/org/apache/httpcomponents/httpcore/4.4.13/httpcore-4.4.13.jar
[ERROR] urls[12] = file:/home/workplace/.m2/repository/commons-codec/commons-codec/1.11/commons-codec-1.11.jar
[ERROR] urls[13] = file:/home/workplace/.m2/repository/org/springframework/spring-core/5.3.14/spring-core-5.3.14.jar
[ERROR] urls[14] = file:/home/workplace/.m2/repository/org/springframework/spring-jcl/5.3.14/spring-jcl-5.3.14.jar
[ERROR] urls[15] = file:/home/workplace/.m2/repository/org/tomlj/tomlj/1.0.0/tomlj-1.0.0.jar
[ERROR] urls[16] = file:/home/workplace/.m2/repository/org/antlr/antlr4-runtime/4.7.2/antlr4-runtime-4.7.2.jar
[ERROR] urls[17] = file:/home/workplace/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar
[ERROR] urls[18] = file:/home/workplace/.m2/repository/org/springframework/boot/spring-boot-loader-tools/2.5.8/spring-boot-loader-tools-2.5.8.jar
[ERROR] urls[19] = file:/home/workplace/.m2/repository/org/apache/maven/shared/maven-common-artifact-filters/3.2.0/maven-common-artifact-filters-3.2.0.jar
[ERROR] urls[20] = file:/home/workplace/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.19/plexus-interpolation-1.19.jar
[ERROR] urls[21] = file:/home/workplace/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.5.5/plexus-component-annotations-1.5.5.jar
[ERROR] urls[22] = file:/home/workplace/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
[ERROR] urls[23] = file:/home/workplace/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[ERROR] urls[24] = file:/home/workplace/.m2/repository/commons-io/commons-io/2.6/commons-io-2.6.jar
[ERROR] urls[25] = file:/home/workplace/.m2/repository/org/eclipse/aether/aether-util/0.9.0.M2/aether-util-0.9.0.M2.jar
[ERROR] urls[26] = file:/home/workplace/.m2/repository/org/sonatype/sisu/sisu-guice/3.1.0/sisu-guice-3.1.0-no_aop.jar
[ERROR] urls[27] = file:/home/workplace/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar
[ERROR] urls[28] = file:/home/workplace/.m2/repository/org/eclipse/sisu/org.eclipse.sisu.inject/0.0.0.M5/org.eclipse.sisu.inject-0.0.0.M5.jar
[ERROR] urls[29] = file:/home/workplace/.m2/repository/org/apache/maven/shared/maven-shared-utils/3.3.3/maven-shared-utils-3.3.3.jar
[ERROR] urls[30] = file:/home/workplace/.m2/repository/org/codehaus/plexus/plexus-utils/3.2.1/plexus-utils-3.2.1.jar
[ERROR] urls[31] = file:/home/workplace/.m2/repository/org/sonatype/plexus/plexus-build-api/0.0.7/plexus-build-api-0.0.7.jar
[ERROR] urls[32] = file:/home/workplace/.m2/repository/org/apache/maven/plugins/maven-shade-plugin/3.2.4/maven-shade-plugin-3.2.4.jar
[ERROR] urls[33] = file:/home/workplace/.m2/repository/org/apache/maven/shared/maven-artifact-transfer/0.12.0/maven-artifact-transfer-0.12.0.jar
[ERROR] urls[34] = file:/home/workplace/.m2/repository/org/ow2/asm/asm/8.0/asm-8.0.jar
[ERROR] urls[35] = file:/home/workplace/.m2/repository/org/ow2/asm/asm-commons/8.0/asm-commons-8.0.jar
[ERROR] urls[36] = file:/home/workplace/.m2/repository/org/ow2/asm/asm-tree/8.0/asm-tree-8.0.jar
[ERROR] urls[37] = file:/home/workplace/.m2/repository/org/ow2/asm/asm-analysis/8.0/asm-analysis-8.0.jar
[ERROR] urls[38] = file:/home/workplace/.m2/repository/org/jdom/jdom2/2.0.6/jdom2-2.0.6.jar
[ERROR] urls[39] = file:/home/workplace/.m2/repository/org/apache/maven/shared/maven-dependency-tree/3.0.1/maven-dependency-tree-3.0.1.jar
[ERROR] urls[40] = file:/home/workplace/.m2/repository/org/vafer/jdependency/2.4.0/jdependency-2.4.0.jar
[ERROR] urls[41] = file:/home/workplace/.m2/repository/org/ow2/asm/asm-util/8.0/asm-util-8.0.jar
[ERROR] urls[42] = file:/home/workplace/.m2/repository/com/google/guava/guava/28.2-android/guava-28.2-android.jar
[ERROR] urls[43] = file:/home/workplace/.m2/repository/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar
[ERROR] urls[44] = file:/home/workplace/.m2/repository/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
[ERROR] urls[45] = file:/home/workplace/.m2/repository/org/checkerframework/checker-compat-qual/2.5.5/checker-compat-qual-2.5.5.jar
[ERROR] urls[46] = file:/home/workplace/.m2/repository/com/google/errorprone/error_prone_annotations/2.3.4/error_prone_annotations-2.3.4.jar
[ERROR] urls[47] = file:/home/workplace/.m2/repository/com/google/j2objc/j2objc-annotations/1.3/j2objc-annotations-1.3.jar
[ERROR] urls[48] = file:/home/workplace/.m2/repository/org/apache/commons/commons-lang3/3.7/commons-lang3-3.7.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException



##### **Motivation for or Use Case**

wanted to test and deploy a monolithic spring-boot application on  Google app engine

<!-- Explain why this is a bug for you -->

##### **Reproduce the error**

Run `Jhipster gae` for configuring and deploy app

then while building application with following , error occured 

`./mvnw package appengine:deploy -DskipTests -Pgae,prod,prod-gae`

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Version(s)**

7.5.0

##### **JHipster configuration**

##### **JHipster Version(s)**

```
xxxx-xxxxx@0.0.1-SNAPSHOT /home/workplace/base-sample
`-- (empty)

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "baseName": "xxxxxx",
    "jhipsterVersion": "7.5.0",
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": ["cypress", "protractor", "gatling", "cucumber"],
    "blueprints": [],
    "otherModules": [],
    "pages": [],
    "creationTimestamp": 1642259499030,
    "serviceDiscoveryType": "no",
    "reactive": false,
    "authenticationType": "session",
    "packageName": "xxxx.xxxxxx",
    "serverPort": "8080",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": true,
    "databaseType": "sql",
    "devDatabaseType": "h2Memory",
    "prodDatabaseType": "mysql",
    "buildTool": "maven",
    "serverSideOptions": ["websocket:spring-websocket", "enableSwaggerCodegen:true"],
    "websocket": "spring-websocket",
    "enableSwaggerCodegen": true,
    "clientFramework": "angularX",
    "withAdminUi": true,
    "clientTheme": "lux",
    "clientThemeVariant": "dark",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "rememberMeKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "devServerPort": 4200,
    "clientPackageManager": "npm",
    "languages": [
      "en",
      "al",
      "ar-ly",
      "hy",
      "by",
      "bn",
      "bg",
      "ca",
      "zh-cn",
      "zh-tw",
      "hr",
      "cs",
      "da",
      "nl",
      "et",
      "fa",
      "fi",
      "fr",
      "gl",
      "de",
      "el",
      "hi",
      "hu",
      "in",
      "it",
      "ja",
      "ko",
      "mr",
      "my",
      "pl",
      "pt-br",
      "pt-pt",
      "pa",
      "ro",
      "ru",
      "sk",
      "sr",
      "si",
      "es",
      "sv",
      "tr",
      "ta",
      "te",
      "th",
      "ua",
      "uz-Cyrl-uz",
      "uz-Latn-uz",
      "vi"
    ],
    "cypressCoverage": true,
    "searchEngine": false,
    "messageBroker": false,
    "enableGradleEnterprise": false,
    "gcpProjectId": "xxxxx-xxxxxx",
    "gcpCloudSqlInstanceName": "xxxxxx-xxxxxx:us-central1:spotgarage",
    "gcpCloudSqlUserName": "root",
    "gcpCloudSqlDatabaseName": "xxxxx",
    "gaeServiceName": "xxx-xxxxx",
    "gaeLocation": "us-central",
    "gaeInstanceClass": "F2",
    "gaeScalingType": "automatic",
    "gaeMinInstances": "0",
    "gaeMaxInstances": "2",
    "gaeCloudSQLInstanceNeeded": "Y"
  }
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>


##### **Environment and Tools**

openjdk version "11.0.13" 2021-10-19
OpenJDK Runtime Environment (build 11.0.13+8-Ubuntu-0ubuntu1.20.04)
OpenJDK 64-Bit Server VM (build 11.0.13+8-Ubuntu-0ubuntu1.20.04, mixed mode, sharing)

git version 2.25.1

node: v14.17.1

npm: 6.14.13

Docker version 20.10.12, build e91ed57

docker-compose version 1.29.2, build 5becea4c


No change to package.json was detected. No package manager install will be executed.


<!--
To provide all information we need, you should run `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
---
If you have a JDL please wrap it in below structure
  <details>
  <summary>JDL definitions</summary>
  <pre>
     JDL content here
  </pre>
  </details>
-->

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->

##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

- [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/gae/index.js
+++ b/generators/gae/index.js
@@ -789,7 +789,7 @@ module.exports = class extends BaseBlueprintGenerator {
       addDependencies() {
         if (this.abort) return;
         if (this.buildTool === MAVEN) {
-          this.addMavenDependency('org.springframework.boot.experimental', 'spring-boot-thin-layout', '1.0.23.RELEASE');
+          this.addMavenDependency('org.springframework.boot.experimental', 'spring-boot-thin-layout', '1.0.28.RELEASE');
         }
         if (this.gaeCloudSQLInstanceNeeded === 'N') return;
         if (this.prodDatabaseType === MYSQL || this.prodDatabaseType === MARIADB) {

